src/components/Map.tsx

```diff
@@ -0,0 +1,183 @@
+      return {
+        'type': 'Feature',
+        'geometry': {
+          'type': 'Polygon',
+          'coordinates': coords
+        }
+      };
+    };
+    this.updateWaypointLine = (coords: any[] = []) => {
+      return {
+        'type': 'Feature',
+        'properties': {},
+        'geometry': {
+          'type': 'LineString',
+          'coordinates': coords
+        }
+      };
+    };
+    this.updateOdlcMarkers = (markers: any[]) => {
+      return {
+        'type': 'FeatureCollection',
+        'features': markers.map((marker: any) => {
+          return {
+            'type': 'Feature',
+            'geometry': {
+              'type': 'Point',
+              'coordinates': marker.pos
+            },
+            'properties': {
+              'title': marker.description,
+              'icon': 'monument'
+            }
+          }
+        })
+      };
+    };
+  }
+
+  componentDidMount() {
+    const map = new mapboxgl.Map({
+      container: this.mapContainer,
+      style: 'mapbox://styles/mapbox/satellite-v9',
+      center: [this.state.lng, this.state.lat],
+      zoom: this.state.zoom
+    });
+    this.map = map;
+    map.on('move', () => {
+      this.setState({
+        lng: map.getCenter().lng.toFixed(4),
+        lat: map.getCenter().lat.toFixed(4),
+        zoom: map.getZoom().toFixed(2)
+      });
+    });
+    map.on('load', () => {
+      this.planeMarker.addTo(map);
+      this.dropMarker.addTo(map);
+      map.addSource('flyzone', {
+        'type': 'geojson',
+        'data': this.updateFlyzonePoly([[]])
+      });
+      map.addLayer({
+        'id': 'flyzone',
+        'type': 'fill',
+        'source': 'flyzone',
+        'layout': {},
+        'paint': {
+          'fill-color': '#088',
+          'fill-opacity': 0.8
+        }
+      });
+      map.addSource('waypoints', {
+        'type': 'geojson',
+        'data': this.updateWaypointLine([])
+      });
+      map.addLayer({
+        'id': 'waypoints',
+        'type': 'fill',
+        'source': 'waypoints',
+        'layout': {}
+      });
+      map.addSource('odlcs', {
+        'type': 'geojson',
+        'data': this.updateOdlcMarkers([])
+      });
+      map.addLayer({
+        'id': 'odlcs',
+        'type': 'symbol',
+        'source': 'odlcs',
+        'layout': {
+          'icon-image': ['concat', ['get', 'icon'], '-15'],
+          'text-field': ['get', 'title'],
+          'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
+          'text-offset': [0, 0.6],
+          'text-anchor': 'top'
+        }
+      });
+    });
+    window.addEventListener('resize', () => {
+      this.setState({ width: window.innerWidth, height: window.innerHeight });
+    });
+  }
+
+  render() {
+    let { width, height } = this.state;
+    return (
+      <div className="map-wrapper">
+        <ServicesContext.Consumer>
+          {({ telemetry, mission, odlcs }) => {
+            if (telemetry) {
+              // @ts-ignore
```

Comment:
What happened here

Reply:
The ts-ignore? The mapbox types are a lil broken so it doesn't like the next line.